### DISCUSSION TITLE
Consider using RSLint instead of ESLint

### DISCUSSION THREAD

**Discussion by OmerWow** [vercel/next.js#35400] [Ideas]
### Describe the feature you'd like to request

Hi,

So basically it's nothing new, Next.js already includes built-in linting, which works great, but considering recent changes from JS based tooling to Rust based tooling, I think a huge performance improvement can be made here too.

### Describe the solution you'd like

My proposal is to replace ESLint with RSLint, which is a linter written in Rust and it is 8x faster on a rather small 11k LOC project (0.05 sec vs 0.4 sec), and an incredible 113x times faster on a medium 42k LOC(0.4 sec vs 47 seconds)!.
I can only assume the performance difference are even crazier the larger the project is.

### Describe alternatives you've considered

Well, none.
The best alternative is ESLint which you already use.

The main problem I can see while implementing RSLint is writing the custom rules you wrote for ESLint again, but this time in Rust.

**Comment by eolme** [comment]
From README:

<img width="709" alt="image" src="https://user-images.githubusercontent.com/11076888/158861874-0993cda4-39d8-4a34-af4a-a4311f54ebab.png">

**Reply by eolme** [reply]
And it looks dead - last release 06 Oct 2021
